drivers/XFSSR.py

```diff
@@ -0,0 +1,248 @@
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation; version 2.1 only.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# XFSSR: Based on local-file storage repository, mounts xfs partition
+
+import SR, SRCommand, FileSR, util, lvutil, scsiutil
+
+import os
+import xs_errors
+import vhdutil
+from lock import Lock
+from constants import EXT_PREFIX
+
+CAPABILITIES = ["SR_PROBE","SR_UPDATE", "SR_SUPPORTS_LOCAL_CACHING", \
+                "VDI_CREATE","VDI_DELETE","VDI_ATTACH","VDI_DETACH", \
+                "VDI_UPDATE","VDI_CLONE","VDI_SNAPSHOT","VDI_RESIZE","VDI_MIRROR", \
+                "VDI_GENERATE_CONFIG",                                \
+                "VDI_RESET_ON_BOOT/2","ATOMIC_PAUSE", "VDI_CONFIG_CBT",
+                "VDI_ACTIVATE", "VDI_DEACTIVATE"]
+
+CONFIGURATION = [ [ 'device', 'local device path (required) (e.g. /dev/sda3)' ] ]
+
+DRIVER_INFO = {
+    'name': 'Local XFS VHD',
+    'description': 'SR plugin which represents disks as VHD files stored on a local XFS filesystem, created inside an LVM volume',
+    'vendor': 'Vates SAS',
+    'copyright': '(C) 2019 Vates SAS',
+    'driver_version': '1.0',
+    'required_api_version': '1.0',
+    'capabilities': CAPABILITIES,
+    'configuration': CONFIGURATION
+    }
+
+DRIVER_CONFIG = {"ATTACH_FROM_CONFIG_WITH_TAPDISK": True}
+
+
+class XFSSR(FileSR.FileSR):
+    DRIVER_TYPE = 'xfs'
+
+    """XFS Local file storage repository"""
+    def handles(srtype):
+        return srtype == XFSSR.DRIVER_TYPE
+    handles = staticmethod(handles)
+
+    def load(self, sr_uuid):
+        if not self._is_xfs_available():
+            raise xs_errors.XenError(
+                'SRUnavailable',
+                opterr='xfsprogs is not installed'
+            )
+
+        self.ops_exclusive = FileSR.OPS_EXCLUSIVE
+        self.lock = Lock(vhdutil.LOCK_TYPE_SR, self.uuid)
+        self.sr_vditype = SR.DEFAULT_TAP
+        if not self.dconf.has_key('device') or not self.dconf['device']:
+            raise xs_errors.XenError('ConfigDeviceMissing')
+
+        self.root = self.dconf['device']
+        for dev in self.root.split(','):
+            if not self._isvalidpathstring(dev):
+                raise xs_errors.XenError('ConfigDeviceInvalid', \
+                      opterr='path is %s' % dev)
+        self.path = os.path.join(SR.MOUNT_BASE, sr_uuid)
+        self.vgname = EXT_PREFIX + sr_uuid
```

Comment:
This looks wrong but it's not from you and is too late (users already have created SRs with the experimental driver)

Reply:
Yes, I suppose we can open a low-priority issue concerning this naming.